### DISCUSSION TITLE
Why app loads multiple times?

### DISCUSSION THREAD

**Discussion by vnmtvlv** [nuxt/nuxt#16491] [General]
Changed just app.vue in this template https://github.com/nuxt/starter/tree/v3

<img width="467" alt="image" src="https://user-images.githubusercontent.com/97898172/199104843-3b5a36dc-2393-4ee6-817c-86ea66164da3.png">
<img width="290" alt="image" src="https://user-images.githubusercontent.com/97898172/199104949-8a365af9-0e7f-4606-9df2-d257fb343726.png">

Why useAsyncData and Setup called twice?
Reproduction https://stackblitz.com/edit/nuxt-starter-vd6bno?file=app.vue

**Comment by vnmtvlv** [comment]
Even if we use https://countapi.xyz/ for counting...
app.vue calls multiple times on server side
<img width="662" alt="image" src="https://user-images.githubusercontent.com/97898172/199112136-5d5fdf85-9c7c-44c3-b7ee-b60ba1ff3fcc.png">

**Comment by vnmtvlv** [comment]
If we have just almost empty app.vue it is called 2-3 times ( why?
<img width="202" alt="image" src="https://user-images.githubusercontent.com/97898172/199112488-8c5f1a0b-2cca-4b6b-a306-af88d41bc618.png">

**Comment by danielroe** [comment]
Probably your browser is making more than one request. Possibly a favicon request.

**Reply by vnmtvlv** [reply]
hm, so it triggers ssr rendering?

**Reply by danielroe** [reply]
try logging the route and see what request is triggering it. but yes, if you don't have a file in your project that matches it, and you only have an `app.vue` (no `pages/` directory) then _every_ request will be rendered with `app.vue`, even ones like `favicon.ico` or `robots.txt`.

**Reply by vnmtvlv** [reply]
okay, so its better not to call any mutation (to external storage) in useAsyncData or useFetch ?
